### PR TITLE
feat: Config option for cursor hiding

### PR DESCRIPTION
**Pull request by yzy-1**
implement #542 

Tested OK under the latest niri unstable.

### CODE REVIEW

File: settings.nix [hunk 1]
```diff
@@ -1192,6 +1192,24 @@ with docs.lib; {
                 This will also set the XCURSOR_SIZE environment variable for all spawned processes.
               '';
             };
+          hide-on-key-press =
+            optional types.bool false
+            // {
+              description = ''
+                ${unstable-note}
+
+                If set, hides the cursor when pressing a key on the keyboard.
```
Comment (sodiboo) [inline settings.nix:1201]: This is copied from niri wiki, yea? In that case, this node is either missing or "set". In our case, there's a slight semantic difference in that it is always "present" and set to false or true; there is no "unset" state for this option, and i think the wording ought to reflect that
```suggestion
                Whether to hide the cursor when a key is pressed on the keyboard.
```

**Comment by sodiboo** [comment]
Could you drop the second merge commit? This merges cleanly without it, and i don't see why it's part of the commit history in this PR.

Other than that and a minor wording change, this looks good to me!

**Comment by sodiboo** [comment]
- Closes #542

###### (this is magic github syntax so it knows the issue is related)

**Comment by yzy-1** [comment]
All done.

**Pull request merged**